File: TriviaServer/TriviaServer/GetQuestionResponse.cpp
#include "GetQuestionResponse.h"

GetQuestionResponse::GetQuestionResponse(int status, string question, vector<string> answers)
{
	this->messageCode = 81;
	this->_status = status;
	this->_question = question;
	this->_answers = answers;
}

nlohmann::json GetQuestionResponse::castToJson() const
{
	return nlohmann::json{ {"status" , _status}, {"question", this->_question}, {"answers", castAnswersToString()} };
}


string GetQuestionResponse::castAnswersToString() const
{
	string str = "";

	vector<string> temp = this->_answers;

	for (vector<string>::iterator it = temp.begin(); it != temp.end(); it++)
	{
		str += *it;
		str += "*";
	}

	return str.substr(0, str.length() - 1);
}

File: TriviaWPFClient/TriviaClient/GetQuestionResponse.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace TriviaClient
{
    class GetQuestionResponse : Response
    {
        public const int MESSAGE_CODE = 81;
        public int status { get; set; }
        public string question { get; set; }
        public string[] answers { get; set; }
        GetQuestionResponse() : base(MESSAGE_CODE){}
        
    }
}
